maliput_sim/export/mcap.py

```diff
@@ -0,0 +1,207 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2023, Woven by Toyota
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from typing import List
+
+from maliput_sim.export.base import Exporter
+from maliput_sim.simulation import Simulation
+from maliput_sim.utils.obj import get_obj_description_from_road_network
+
+from foxglove_schemas_protobuf.Color_pb2 import Color
+from foxglove_schemas_protobuf.KeyValuePair_pb2 import KeyValuePair
+from foxglove_schemas_protobuf.Pose_pb2 import Pose
+from foxglove_schemas_protobuf.Quaternion_pb2 import Quaternion
+from foxglove_schemas_protobuf.Vector3_pb2 import Vector3
+from foxglove_schemas_protobuf.SceneUpdate_pb2 import SceneUpdate
+from google.protobuf.duration_pb2 import Duration
+from mcap_protobuf.writer import Writer
+
+
+class MCAP(Exporter):
+    """A Foxglove visualization. This class dumps the simulation states into a mcap file that can be
+    visualized using Foxglove Studio."""
+
+    _DEFAULT_SIMULATION_OUTPUT_FILE_PATH = "simulation.mcap"
+
+    _AGENT_BOX_SIZE = [4, 2, 1.5]
```

Comment:
Here and there, consider initializing the actual components with their types instead of lists. It'll make the code way more straightforward and you'll save from an unnecessary type conversion.

Reply:
Done.